### ISSUE TITLE
version.locked? method returns incorrect value of true for unlocked version

### ISSUE THREAD

**Issue by jondade** [fastly/fastly-ruby#47]
When calling Version.locked? on an unlocked / freshly cloned version the response is always true regardless of actual state.

The only time this is not true is if the version was created / cloned by this instance of Fastly.

**Comment by eddieroger** [comment]
I just noticed that some of my upload scripts are behaving the same way, where getting the latest version shows that it's unlocked, but when I inspected the `service.versions` response, it appears to be locked. For example: 

```
2.2.3 :025 > service.versions.last.locked?
 => false
2.2.3 :026 > service.versions.last.number
 => 387
```

But when I look at `service.versions`, I see the last version is marked as locked: 

```
{"testing"=>false, "locked"=>true, "number"=>387, "active"=>true, "service_id"=>"[my_service]", "staging"=>false, "created_at"=>"2016-05-26T03:33:51+00:00", "deleted_at"=>nil, "comment"=>"", "updated_at"=>"2016-05-26T03:34:08+00:00", "deployed"=>false}
```

Haven't done more digging than that, but I'm seeing the same behavior and am glad it's not just me. Creating a new clone in the Fastly UI was how I got the upload scripts to work again for me as well.

**Comment by eddieroger** [comment]
One more thing (I promise to start coalescing my updates 😄 ) - when calling the API directly, the version is returned as locked. I looked through the history of Version.rb, and right before this ticket was made, a commit removed `:locked` from attr_accessor, which I think is causing problems with the new logic more recently added. I went in to my local gem source and added `:locked` back to the attr_accessor line, and sure enough `v.locked?` worked for me again. 

Here's the commit where it was removed: 

https://github.com/fastly/fastly-ruby/commit/ebbfb3c160a0d21ff349699423bec7b685648dbb

And my `irb` after I modified my local gem source:

```
2.2.3 :021 > v.locked?
 => true
2.2.3 :022 > v.number
 => 389
```

Corresponding API call: 

```
        {
            "active": true,
            "comment": "",
            "created_at": "2016-05-27T12:49:34+00:00",
            "deleted_at": null,
            "deployed": false,
            "locked": true,
            "number": 389,
            "service_id": "[my service]",
            "staging": false,
            "testing": false,
            "updated_at": "2016-05-27T12:50:08+00:00"
        }
```

**Comment by lanej** [comment]
Fixed by #83 #81
